client/client.cpp

```diff
@@ -0,0 +1,192 @@
+        mainLayout = new QGridLayout(groupBox);
+        outerHorizontalLayout->addWidget(groupBox);
+        outerHorizontalLayout->addItem(new QSpacerItem(0, 0, QSizePolicy::MinimumExpanding, QSizePolicy::Ignored));
+        outerVerticalLayout->addLayout(outerHorizontalLayout);
+        outerVerticalLayout->addItem(new QSpacerItem(0, 0, QSizePolicy::Ignored, QSizePolicy::MinimumExpanding));
+    }
+    else
+    {
+        mainLayout = new QGridLayout(this);
+    }
+    mainLayout->addWidget(hostLabel, 0, 0);
+    mainLayout->addWidget(hostCombo, 0, 1);
+    mainLayout->addWidget(portLabel, 1, 0);
+    mainLayout->addWidget(portLineEdit, 1, 1);
+    mainLayout->addWidget(statusLabel, 2, 0, 1, 2);
+    mainLayout->addWidget(buttonBox, 3, 0, 1, 2);
+    setWindowTitle(QGuiApplication::applicationDisplayName());
+    portLineEdit->setFocus();
+
+}
+void Client::requestNew()
+{
+    getButton->setEnabled(false);
+    tcpSocket->abort();
+
+    tcpSocket->connectToHost(hostCombo->currentText(),
+                             portLineEdit->text().toInt());
+}
+
+void Client::displayError(QAbstractSocket::SocketError socketError)
+{
+    switch (socketError) {
+    case QAbstractSocket::RemoteHostClosedError:
+        break;
+    case QAbstractSocket::HostNotFoundError:
+        QMessageBox::information(this, tr(" Client"),
+                                 tr("The host was not found. Please check the "
+                                    "host name and port settings."));
+        break;
+    case QAbstractSocket::ConnectionRefusedError:
+        QMessageBox::information(this, tr(" Client"),
+                                 tr("The connection was refused by the peer. "
+                                    "Make sure the  server is running, "
+                                    "and check that the host name and port "
+                                    "settings are correct."));
+        break;
+    default:
+        QMessageBox::information(this, tr(" Client"),
+                                 tr("The following error occurred: %1.")
+                                 .arg(tcpSocket->errorString()));
+    }
+
+    getButton->setEnabled(true);
+}
+
+void Client::read()
+{
+    in >> list;
+
+    manager = new QNetworkAccessManager(this);
+
+
+    for(int i = 0; i < list.size(); ++i)
+    {
+
+        QString l = list[i];
+        QUrl url = QUrl::fromUserInput( l);
+        QNetworkReply *reply = manager->get(QNetworkRequest(url));
+
+        in.startTransaction();
+
+        connect(reply,&QNetworkReply::finished,this,&Client::fileHandler);
+
+
+    }
+
+}
+
+
+
+
+
+
+void Client::fileHandler()
+{
+    auto reply = dynamic_cast<QNetworkReply*>(sender());
+    if(!reply)
+    {
+        return;
+    }
+
+    QString name = QString("imageFiles_%1.png").arg(QDateTime::currentMSecsSinceEpoch());
+
+    QFile destinationFile(name);
+
+    if(destinationFile.open(QFile::WriteOnly))
+    {
+
+        auto data = reply->readAll();
```

Comment:
check for errors in reply first, maybe reply did not properly return the file body.